src/Cli/Microsoft.TemplateEngine.Cli/TemplatePackageCoordinator.cs

```diff
@@ -206,20 +167,19 @@ internal async Task<NewCommandStatus> EnterInstallFlowAsync(InstallCommandArgs a
             var managedSourceProvider = _templatePackageManager.GetBuiltInManagedProvider(InstallationScope.Global);
             List<InstallRequest> installRequests = new();
 
-            foreach (string installArg in args.TemplatePackages)
+            foreach (var packageIdentifier in args.TemplatePackages)
             {
-                string[] split = installArg.Split(["::"], StringSplitOptions.RemoveEmptyEntries).SelectMany(arg => arg.Split('@', StringSplitOptions.RemoveEmptyEntries)).ToArray();
-                string identifier = split[0];
-                string? version = split.Length > 1 ? split[1] : null;
-
-                if (installArg.Contains("::"))
+                if (packageIdentifier is PackageIdentifier id)
                 {
-                    Reporter.Output.WriteLine(string.Format(LocalizableStrings.Colon_Separator_Deprecated, split[0], split.Length > 1 ? split[1] : string.Empty).Yellow());
+                    installRequests.Add(new InstallRequest(id.Package.Id, version: id.Package.Version?.ToString(), details: details, force: args.Force));
                 }
-
-                foreach (string expandedIdentifier in InstallRequestPathResolution.ExpandMaskedPath(identifier, _engineEnvironmentSettings))
+                else if (packageIdentifier is FileBasedTemplateIdentifier fileIdentifier)
+                {
+                    installRequests.Add(new InstallRequest(fileIdentifier.File.FullName, version: null, details: details, force: args.Force));
+                }
+                else
                 {
-                    installRequests.Add(new InstallRequest(expandedIdentifier, version, details: details, force: args.Force));
+                    throw new ArgumentException($"Unknown template package identifier type: {packageIdentifier.GetType().Name}");
                 }
```

Comment:
here's the pivot on the different variants - note that since the C# compiler doesn't do exhaustivity-checks we need a catch-all clause. this clause can be unlocalized because we should only hit it during development time (we shouldn't introduce variants without expanding this)